### PR TITLE
[CI] Use `cargo-contract` from container

### PR DESCRIPTION
**Pull request by cmichi**
Can be merged as a follow-up to https://github.com/paritytech/scripts/pull/241 (which closes https://github.com/paritytech/ci_cd/issues/53).

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -242,27 +235,13 @@ examples-contract-build:
...
   script:
-    - *update-cargo-contract
+    - cargo contract -V
     - for example in examples/*/; do
         pushd $example &&
         cargo contract build &&
         popd;
       done
 
-examples-generate-metadata:
```
Comment (TriplEight) [inline .gitlab-ci.yml:252]: why is this removed within this PR?

Comment (cmichi) [reply .gitlab-ci.yml:252]: We removed this command `cargo contract generate-metadata` in the upcoming v0.8.0 release, it's no found as a flag under `cargo contract build --metadata-only`.

**Review by TriplEight — APPROVED** [review]
LGTM

**Review by Robbepop — APPROVED** [review]
